src/server.c

```diff
@@ -203,10 +203,17 @@ int pool_pool_mode(PgPool *pool)
 
 int pool_pool_size(PgPool *pool)
 {
-	if (pool->db->pool_size < 0)
-		return cf_default_pool_size;
-	else
+	/* both database and user max pool limits are not configured */
```

Comment:
The most restrictive of both `[databases]` and `[pools]` `pool_size` will be enforced.